core/services/balance_monitor.go

```diff
@@ -141,3 +94,48 @@ func (bm *balanceMonitor) GetEthBalance(address gethCommon.Address) *assets.Eth
 	defer bm.ethBalancesMtx.RUnlock()
 	return bm.ethBalances[address]
 }
+
+type worker struct {
+	bm *balanceMonitor
+}
+
+func (w *worker) Work() {
+	keys, err := w.bm.store.Keys()
+	if err != nil {
+		logger.Error("BalanceMonitor: error getting keys", err)
+	}
+
+	var wg sync.WaitGroup
+
+	wg.Add(len(keys))
+	for _, key := range keys {
+		go func(k models.Key) {
+			w.checkAccountBalance(k)
+			wg.Done()
+		}(key)
+	}
+	wg.Wait()
+}
+
+const ethFetchTimeout = 2 * time.Second
```

Comment:
We may wish to make this much longer now we no longer need to complete within one head.

Say, 15s or so.